cli_tools/osconfig_agent/main.go

```diff
@@ -42,22 +101,42 @@ func strIn(s string, ss []string) bool {
 	return false
 }
 
-func main() {
-	flag.Parse()
-	patchInit()
-
-	ctx := context.Background()
-
+func run(ctx context.Context) {
 	client, err := osconfig.NewClient(ctx, option.WithEndpoint(*endpoint), option.WithCredentialsFile(*oauth))
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("NewClient Error:", err)
 	}
 
-	res, err := lookupConfigs(ctx, client, *resource)
+	res, err := getResource(*resource)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("getResource error:", err)
 	}
-	patchManager(res.PatchPolicies)
 
-	//runUpdates()
+	patchInit()
+	ticker := time.NewTicker(interval)
+	for {
+		res, err := lookupConfigs(ctx, client, res)
+		if err != nil {
+			log.Println("ERROR:", err)
+		} else {
+			runPackageConfig(res)
+			patchManager(res.PatchPolicies)
+		}
+
+		select {
+		case <-ticker.C:
```

Comment:
Just curious, why do you use a `Ticker` here but `Sleep` for retrying the resource lookup?

Reply:
Sleep just sleeps, this waist for the first channel to send something. In the normal case the ticker will just tick and the loop will continue, but if the service is told to restart this will break the service loop. This is mostly needed for Windows services but in the future we may add more cleanup that needs to happen here.